app/views/registration/_form.erb

```diff
@@ -0,0 +1,71 @@
+<% title _('Register Host') %>
+<%= form_tag({controller: 'registration', action: 'create' }, method: :post, class: 'form-horizontal well') do %>
+  <div class='form-group'>
+    <label class='col-md-2 control-label'><%= _('Organization') %></label>
+    <div class='col-md-4'>
+      <%= select_tag 'organization', options_from_collection_for_select(@organizations, :id, :name, params[:organization]), class: 'form-control' %>
+    </div>
+  </div>
+  <div class='form-group'>
+    <label class='col-md-2 control-label'><%= _('Location') %></label>
+    <div class='col-md-4'>
+      <%= select_tag 'location', options_from_collection_for_select(@locations, :id, :name, params[:location]), class: 'form-control' %>
+    </div>
+  </div>
+  <div class='form-group'>
+    <label class='col-md-2 control-label'><%= _('Host Group') %></label>
+    <div class='col-md-4'>
+      <%= select_tag 'host_group', options_from_collection_for_select(@host_groups, :id, :name, params[:host_group]), class: 'form-control', include_blank: '' %>
+    </div>
+  </div>
+  <div class='form-group'>
+    <label class='col-md-2 control-label'><%= _('Operating System') %></label>
+    <div class='col-md-4'>
+      <%= select_tag 'operating_system', options_from_collection_for_select(@operating_systems, :id, :title, params[:operating_system]), class: 'form-control', include_blank: '' %>
+    </div>
+  </div>
+  <div class='form-group'>
+    <label class='col-md-2 control-label'><%= _('Proxy') %></label>
+    <div class='col-md-4'>
+      <%= select_tag 'smart_proxy', options_from_collection_for_select(@smart_proxies, :id, :name, params[:smart_proxy]), class: 'form-control', include_blank: '' %>
+    </div>
+  </div>
+  <div class='form-group'>
+    <label class='col-md-2 control-label'><%= _('Token lifetime (hours)') %></label>
+    <div class='col-md-4'>
+      <%= number_field_tag 'jwt_expiration', 4, class: 'form-control', min: 1 %>
+    </div>
+  </div>
+  <% @plugin_params.each do |pr| %>
+    <div class='form-group '>
+      <label class='col-md-2 control-label'><%= pr.to_s.titleize %></label>
+      <div class='col-md-4'>
+        <%= text_field_tag pr, params[pr], class: 'form-control' %>
```

Comment:
This does not allow any advanced customization of the form. E.g. Katello certainly will want to add activation key selector, text field is not the best user experience. I'm fine with this for an initial version, but I think it would be good to add a pagelet instead and let plugins define whatever form field they find suitable.